File: BackEnd/retail.demo/src/main/java/com/retail/demo/service/OrderService.java
package com.retail.demo.service;

import com.retail.demo.entity.InventoryItemEntity;
import com.retail.demo.entity.OrderedItemEntity;
import com.retail.demo.model.OrderedItem;
import com.retail.demo.model.UserOrder;

import java.util.List;

public interface OrderService {
    UserOrder createOrder(UserOrder userOrder);
    UserOrder updateOrder(Long id, UserOrder userOrder);
    UserOrder getOrder(Long id);
    List<UserOrder> getOrdersByUserId(Long id);
    boolean deleteOrder(Long id);
    List<UserOrder> getAllOrders();
    OrderedItem addItemToOrder(Long orderId, InventoryItemEntity inventoryItemEntity);
    boolean removeItemFromOrder(Long orderId, OrderedItemEntity inventoryItemEntity);
    List<OrderedItem> getAllItemsInOrder(Long orderId);
    boolean isDaoEmpty();
    UserOrder getLastOrder();

    UserOrder setViewingOrder(UserOrder userOrder);

    UserOrder getViewingOrder();

}

File: FrontEnd/retail-web-app/src/service/OrderService.js
import axios from 'axios';
const API_URL = 'http://localhost:9080/OrderApi/v1/orders';

class OrderService {
    
        getAllOrders() {
            return axios.get(API_URL);
        }
    
        getOrderById(orderId) {
            return axios.get(API_URL + '/' + orderId);
        }

        getOrdersByUserId(userId) {
            return axios.get(API_URL + '/user/' + userId);
        }

        getViewingOrder() {
            return axios.get(API_URL + '/viewOrder');
        }

        setViewingOrder(order) {
            return axios.post(API_URL + '/viewOrder', order);
        }
    
        createOrder(order) {
            return axios.post(API_URL, order);
        }
    
        updateOrder(order, orderId) {
            return axios.put(API_URL + '/' + orderId, order);
        }
    
        deleteOrder(orderId) {
            return axios.delete(API_URL + '/' + orderId);
        }
    }

export default new OrderService()